src/coreclr/vm/dllimportcallback.h

```diff
@@ -227,11 +227,29 @@ class UMEntryThunkData
         if (m_pObjectHandle == NULL && m_pManagedTarget == (TADDR)0)
             m_pManagedTarget = m_pMD->GetMultiCallableAddrOfCode();
 
-        m_pUMEntryThunk->SetTargetUnconditional(m_pUMThunkMarshInfo->GetExecStubEntryPoint());
+#ifdef FEATURE_INTERPRETER
+        // HACK: For interpreted stubs we need to ensure that TheUMEntryPrestubWorker runs for every
+        //  unmanaged-to-managed invocation in order to populate the TLS variable every time.
+        PCODE pILStub = m_pUMThunkMarshInfo->GetExecStubEntryPoint();
+        Precode* pStubPrecode = (Precode*)(void*)pILStub;
```

Comment:
What guarantees that this is precode? I think this can be actual code when the code is JITed.

Reply:
I tried using EECodeInfo and it didn't work, so I'm not sure what to do here. If we were to expose the MethodDesc for the stub, we could just query its InterpreterCode directly, but doing that creates an annoying race because of how it populates the il stub using an atomic exchange.

Reply:
EECodeInfo is mean to be used for dealing with IP inside the method code.

NonVirtualEntry2MethodDesc is the best method to use to map entrypoint like we have here to MethodDesc*.. (You may need some fixes from https://github.com/dotnet/runtime/pull/119129/ to make it work.)